.github/workflows/precompiled.yaml

```diff
@@ -93,19 +100,44 @@ jobs:
           IMAGE_NAME: ghcr.io/nvidia/driver
           VERSION: ${COMMIT_SHORT_SHA}
           PRECOMPILED: "true"
-          DIST: signed_ubuntu22.04
+          DIST: signed_${{ matrix.dist }}
         run: |
           source kernel_version.txt && \
-          make DRIVER_VERSIONS=${DRIVER_VERSIONS} DRIVER_BRANCH=${{ matrix.driver-branch }} build-${DIST}-${DRIVER_VERSION}
+          make DRIVER_VERSIONS=${DRIVER_VERSIONS} DRIVER_BRANCH=${{ matrix.driver_branch }} build-${DIST}-${DRIVER_VERSION}
+
+      - name: Save build image as a tar
+        env:
+          DIST: ${{ matrix.dist }}
+          PRIVATE_REGISTRY: "ghcr.io"
+          COMMIT_SHORT_SHA: ${COMMIT_SHORT_SHA}
+        run: |
+          source kernel_version.txt
+          docker images "${PRIVATE_REGISTRY}/nvidia/driver:${{ matrix.driver_branch }}-${KERNEL_VERSION}-${DIST}"
```

Comment:
Question -- is this line required?

Reply:
to build local docker image(   do not push with flag GENERATE_ARTIFACTS="false") for docker save command,  which uploads .tar image as artifacts

Reply:
But why is this needed? Doesn't `docker images` just list the image? Is this to verify the tag exists before attempting to save? If it doesn't exist I would expect the save command to return an error.

Reply:
docker images will list the images, but we need a locally built image for the current code for e2etests.
Also, if we want to e2etest an earlier release from the repo, we need to run docker pull to get the old image. 

**Old workflow :** 

1.   Build local image (current code build) and push image to the repo 
2.   e2e tests  
     a. upgrade kernel 
      b. fetch repo image on target with helm install to run driver tests

**New Workflow:**

**Build Local Image (Current Code Build)**
    a. Build the image
    a. Save the built image as a .tar file using docker save.
    b. Upload the .tar file as an artifact (not to the repo).

**e2e Tests**
    a. Upgrade the kernel on the target.
    b. Download the artifact (the saved .tar image file) from the CI pipeline
    c. Transfer the .tar image to the target using rsync.
    d. Import the .tar image on the target using the ctr command.
    e. Run helm install to execute driver tests.

**Publish or Push Image to Repo**
    a. Download the artifact.
    b. Load the .tar image using docker load.
    c. Push the image to the repo using docker push.

Reply:
Thanks .   removed 

The docker images command is not required (it was previously added for debugging purposes).
I was confused about the command docker images and the make command